cmake/arpackng-config.cmake.in

```diff
@@ -12,17 +12,29 @@
 #   target_include_directories(main INTERFACE PARPACK::PARPACK)
 #   target_link_libraries(main PARPACK::PARPACK)
 
-# Create arpack targets.
-add_library(ARPACK::ARPACK INTERFACE IMPORTED)
-set_target_properties(ARPACK::ARPACK PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_INCLUDEDIR@/arpack-ng@ITF64SUFFIX@")
-set_target_properties(ARPACK::ARPACK PROPERTIES INTERFACE_LINK_DIRECTORIES    "@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@")
-set_target_properties(ARPACK::ARPACK PROPERTIES INTERFACE_LINK_LIBRARIES      "arpack@LIBSUFFIX@@ITF64SUFFIX@")
+if (NOT @BUILD_SHARED_LIBS@)
+	include(CMakeFindDependencyMacro)
+	# Find dependencies
+	if (NOT TARGET BLAS::BLAS)
+		find_dependency(BLAS REQUIRED)
+	endif()
+	if (NOT TARGET LAPACK::LAPACK)
+		find_dependency(LAPACK REQUIRED)
+	endif()
+	if (@ICB@)
+		enable_language(Fortran)
+	endif()
+	if (@MPI@)
+		include(FindMPI)
+		if (NOT TARGET MPI::Fortran)
+			find_dependency(MPI REQUIRED COMPONENTS Fortran)
+		endif()
+	endif()
+endif()
 
-# Create parpack targets.
-set(MPI "@MPI@")
-if (MPI)
-    add_library(PARPACK::PARPACK INTERFACE IMPORTED)
-    set_target_properties(PARPACK::PARPACK PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_INCLUDEDIR@/arpack-ng@ITF64SUFFIX@")
-    set_target_properties(PARPACK::PARPACK PROPERTIES INTERFACE_LINK_DIRECTORIES    "@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@")
-    set_target_properties(PARPACK::PARPACK PROPERTIES INTERFACE_LINK_LIBRARIES      "parpack@LIBSUFFIX@@ITF64SUFFIX@")
-endif ()
+include("${CMAKE_CURRENT_LIST_DIR}/arpackngTargets.cmake")
```

Comment:
@juanjosegarciaripoll: in connection with previous comment, just to understand, a .cmake file is created here, right? what does it contain and why? How is it used by cmake? (I try to improve my knowldge of cmake ! :))

Reply:
As mentioned above, this file describes the two targets that this project create, 'arpack' and 'parpack', creating them as CMake objects with the right values for all properties (include directories, link libraries, link flags, etc). It is done so in a way that the target is immediately available once this file is included from arpackng-config.cmake. It is the modern way of exporting targets.

Reply:
Thanks for clarifications!